setup.py

```diff
@@ -74,6 +74,11 @@ def find_sources():
     platform_specific['define_macros'] = [('WIN32', 1)]
     jni_md_platform = 'win32'
 
+elif sys.platform == 'cygwin':
```

Comment:
Looking at the PyUnicode.  I didn't change it.   I was just cleaning up the warnings that I was getting about redefined macros.   It is likely that the statement that followed should have been conditional.